lib/native/binding.d.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+type NativeHandle = any;
+type StringLike = string
+                | ArrayBuffer
+                | DataView
+                | Int8Array
+                | Uint8Array
+                | Int16Array
+                | Uint16Array
+                | Int32Array
+                | Uint32Array
+                | Float32Array
+                | Float64Array;
+
+/* IO */
+export declare function error_code_to_string(error_code: number): string;
+export declare function error_code_to_name(error_code: number): string;
+export declare function io_logging_enable(log_level: number): void;
+export declare function is_alpn_available(): boolean;
+export declare function io_client_bootstrap_new(): NativeHandle;
+export declare function io_tls_ctx_new(
+    min_tls_version: number,
+    ca_filepath?: StringLike,
```

Comment:
I don't think that's gonna work out for us, considering that we will only read it as a string on the other side for most of these. I think we just need to use string here.

Like, I challenge you to feed a ```Float64Array``` to ```ca_filepath``` :)

Reply:
I wrote these to match the C implementation, not the intent. Anywhere you see `StringLike`, the C impl is calling `aws_byte_buf_init_from_napi`, and `StringLike` is the list of parameters that that function accepts.